lib/trie.ex

```diff
@@ -150,8 +150,9 @@ defmodule KrDict.Trie do
       {%TrieNode{children: children}, match} ->
         children
         |> Enum.flat_map(fn {val, _node} ->
-          [match] ++ do_prefix_lookup(trie, match <> <<val::utf8>>)
+          do_prefix_lookup(trie, match <> <<val::utf8>>)
```

Comment:
I'm not sure that a binary pattern match will work here in all cases?  Is there anything I'm missing?